src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -169,6 +170,59 @@ public void InitializeService(ServiceHost serviceHost)
         }
 
         /// <summary>
+        /// Handles a request to execute a string and return the result
+        /// </summary>
+        internal Task HandleExecuteAndReturnResultRequest(ExecuteAndReturnResultParams executeParams,
+            RequestContext<ExecuteAndReturnResultResult> requestContext)
+        {
+            Func<string, Task> queryCreateFailureAction = message => requestContext.SendError(message);
+            ExecuteStringParams newParams = new ExecuteStringParams
+            {
+                Query = executeParams.QueryString,
+                OwnerUri = executeParams.OwnerUri
+            };
+
+            ResultOnlyContext<ExecuteAndReturnResultResult> newContext = new ResultOnlyContext<ExecuteAndReturnResultResult>(requestContext);
+
+            // handle sending event back when the query completes
+            Query.QueryAsyncEventHandler queryComplete = async q =>
+            {
+                int rowCount = 0;
+                try
+                {
+                    rowCount = Convert.ToInt32(q.Batches[0].ResultSets[0].RowCount);
+                }
+                catch(OverflowException e)
+                {
+                    await requestContext.SendError(e);
```

Comment:
I would recommend checking batch count explicitly first, and sending a clear message over the channel instead of the try\catch exception pattern. That pattern is discouraged in C# since first-chance exceptions can slow debugging (and it's just not something C# seems to like). 

Also, I'd ask what should happen when we have more than 1 batch? In that case should something be done differently?